publishing/lightning-kmp-snapshot-deploy.sh

```diff
@@ -0,0 +1,56 @@
+#!/bin/bash -x
```

Comment:
What is the purpose of this file?

Reply:
To publish snapshot libraries to sonatype so they can be available globally (for other devs/machines, github CI, ...)